### DISCUSSION TITLE
Issues with running PCPCombine in subtraction mode

### DISCUSSION THREAD

**Discussion by malloryprow** [dtcenter/METplus#1364] [Configuration] [answered]
I have two GFS forecast files one for forecast hour 48 and another for 24 hours. The GFS forecast files have a continuous bucket for APCP, so for the 48 hour forecast the accumulation is 0-48 hour accumulation and for the 24 hour forecast the accumulation is 0-24 hour accumulation. I want to get the 24 hour accumulation forecast from forecast 48, so what I need to do is just do f48 - f24.

I'm trying to use the PCPCombine wrapper to do this but I have been running into issues (I am running on WCOSS). Here is what my METplus conf file looks like

> [config]
> MET_INSTALL_DIR = /gpfs/dell2/emc/verification/noscrub/emc.metplus/met/10.0.1
> #### Input
> INPUT_BASE = /gpfs/dell2/ptmp/Mallory.Row/metplus_pcpcombine_subtract_test/input
> FCST_PCP_COMBINE_RUN = True
> OBS_PCP_COMBINE_RUN = False
> FCST_PCP_COMBINE_INPUT_DIR = {INPUT_BASE}
> FCST_PCP_COMBINE_INPUT_TEMPLATE = gfs.t{init?fmt=%Y%m%d%H}z.pgrb2.0p25.f{lead?fmt=%HHH}
> FCST_PCP_COMBINE_INPUT_DATATYPE = GRIB2
> FCST_IS_PROB = False
> #### Output
> OUTPUT_BASE =/gpfs/dell2/ptmp/Mallory.Row/metplus_pcpcombine_subtract_test/output
> LOG_DIR = {OUTPUT_BASE}/logs
> TMP_DIR = {OUTPUT_BASE}/tmp
> FCST_PCP_COMBINE_OUTPUT_DIR = {OUTPUT_BASE}
> FCST_PCP_COMBINE_OUTPUT_TEMPLATE = f{lead?fmt=%HHH}.{init?fmt=%Y%m%d%H}.A{level?fmt=%HHH}hr
> LOG_METPLUS = {LOG_DIR}/MRW_PCPCombine_fcst{MODEL}_Precip_runon{CLOCK_TIME}.log
> METPLUS_CONF = {OUTPUT_BASE}/confs/MRW_GridStat_fcst{MODEL}_Precip_runon{CLOCK_TIME}.conf
> #### METplus application
> LOOP_ORDER = times
> PROCESS_LIST = PCPCombine
> ##### Dates
> LOOP_BY = VALID
> VALID_TIME_FMT = %Y%m%d%H
> VALID_BEG = 2022011812
> VALID_END = 2022011812
> VALID_INCREMENT = 24H
> #### Lead
> LEAD_SEQ = 48H
> ##### Variables
> #### PCPCombine
> PCP_COMBINE_SKIP_IF_OUTPUT_EXISTS = True
> MODEL = gfs
> FCST_PCP_COMBINE_METHOD = SUBTRACT
> FCST_PCP_COMBINE_INPUT_NAMES = APCP
> FCST_PCP_COMBINE_INPUT_ACCUMS = {lead?fmt=%HH}
> FCST_PCP_COMBINE_INPUT_LEVELS = Z0
> FCST_PCP_COMBINE_IS_DAILY_FILE = False
> FCST_PCP_COMBINE_OUTPUT_ACCUM = 24
> FCST_PCP_COMBINE_MIN_FORECAST = 24
> FCST_PCP_COMBINE_MAX_FORECAST = 48
> FCST_PCP_COMBINE_DATA_INTERVAL = 24
> FCST_PCP_COMBINE_TIMES_PER_FILE =
> FCST_PCP_COMBINE_OUTPUT_NAME = APCP_A24
> FCST_PCP_COMBINE_CONSTANT_INIT = True

From that, the MET command it builds to run is 

> /gpfs/dell2/emc/verification/noscrub/emc.metplus/met/10.0.1/exec/pcp_combine -v 2 -subtract /gpfs/dell2/ptmp/Mallory.Row/metplus_pcpcombine_subtract_test/input/gfs.t2022011612z.pgrb2.0p25.f048 'name="APCP_A24";' /gpfs/dell2/ptmp/Mallory.Row/metplus_pcpcombine_subtract_test/input/gfs.t2022011612z.pgrb2.0p25.f024 'name="APCP_A24";' /gpfs/dell2/ptmp/Mallory.Row/metplus_pcpcombine_subtract_test/output/f048.2022011612.A024hr

which fails with

> ERROR  :
> ERROR  : VarInfoGrib2::set_dict() -> unrecognized GRIB2 field abbreviation 'APCP_A24'
> ERROR  :

FCST_PCP_COMBINE_OUTPUT_NAME is being set to "APCP_A24". I'm not sure why that is being used as the input name in the pcp_combine command.

**Comment by georgemccabe** [comment, marked as answer]
Hi Mallory, could you try using METplus 4.1.0-beta2 or higher to see if that fixes the issue? There was a large refactor of the PCPCombine wrapper in that beta release that fixed a lot of issues with it.

**Reply by malloryprow** [reply]
Yup all good now. I had to change FCST_PCP_COMBINE_INPUT_LEVELS = Z0 to FCST_PCP_COMBINE_INPUT_LEVELS = A{lead?fmt=%HH}, and it worked great!

**Reply by georgemccabe** [reply]
Great to hear! To confirm, did you have to use the beta version or were you able to get it to run using 4.0.0?

**Reply by malloryprow** [reply]
Beta. It was 4.1.0-beta5.

**Reply by georgemccabe** [reply]
Great, thanks for confirming!
